### PR TITLE
Improvement/CX-5504 immediate results in country dropdown

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
Following feedback from user testing, some users found it unclear what country options to search for.

# Solution
When user clicks/taps on the search input, show the full list of countries in the dropdown options. As soon as user enters first character, start filtering countries in the list.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [n/a] Has the CHANGELOG been updated? - feature not released yet
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1176-pr-onfido-sdk-ui-onfido.surge.sh

**Review by ethan605 — CHANGES_REQUESTED** [review]

**Review by ethan605 — APPROVED** [review]
LGTM!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1176-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/CountrySelector/style.scss [hunk 1]
```diff
@@ -121,33 +122,35 @@ $_sdk-option-border-bottom: 0;
...
+  background-color: transparent;
+  position: relative;
 }
 
 .custom__input--focused {
   outline-offset: 0;
   box-shadow: inset 0 0 0 2 * $unit;
 }
 
+/* If showAllValues is disabled, change class to .custom__input--default & add @extend %_sdk-caret-icon; */
```
Comment (stefaniacardenas) [inline src/components/CountrySelector/style.scss:140]: When is `showAllValues` disabled?

File: src/components/CountrySelector/style.scss [hunk 2]
```diff
@@ -100,9 +101,9 @@ $_sdk-option-padding: 8 * $unit-small;
 $_sdk-option-border-bottom: 0;
 
 %_sdk-caret-icon {
```
Comment (stefaniacardenas) [inline src/components/CountrySelector/style.scss:103]: Why is it called `sdk-caret-icon` if it's not an icon 🤔

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/CountrySelector/style.scss (hunk 1 above)
Comment (kopijunkie) [reply src/components/CountrySelector/style.scss:140]: Never for us now with this change. Just added the note as a reminder of the relationship to the classes.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1176-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/CountrySelector/style.scss (hunk 2 above)
Comment (stefaniacardenas) [reply src/components/CountrySelector/style.scss]: Sorry, I meant it's not a caret icon ^. It' a magnifying glass, isn't it?

File: src/components/CountrySelector/index.js [hunk 3]
```diff
@@ -115,7 +115,8 @@ class CountrySelection extends Component<Props, State> {
             <Autocomplete
               id="country-search"
               source={this.suggestCountries}
-              minLength={2}
+              showAllValues
+              dropdownArrow={() => <i className={style.dropdownArrow} />}
```
Comment (stefaniacardenas) [inline src/components/CountrySelector/index.js:145]: I would call this class `style.dropdownIcon`

**Review by stefaniacardenas — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/CountrySelector/style.scss (hunk 2 above)
Comment (kopijunkie) [reply src/components/CountrySelector/style.scss]: True. On thinking more about it the most appropriate name for it is `%_sdk-search-icon`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1176-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1176-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**